lib/src/signed_video_h26x_auth.c

```diff
@@ -758,25 +759,19 @@ maybe_validate_gop(signed_video_t *self, h26x_nalu_t *nalu)
       if (item->nalu && item->nalu->is_gop_sei && item->validation_status == 'P') {
         const uint8_t *tlv_data = item->nalu->tlv_data;
         size_t tlv_size = item->nalu->tlv_size;
-        // TODO: tlv_find_tag -> tlv_find_tag_and_decode?
-        const uint8_t *public_key_tag_ptr =
-            tlv_find_tag(self, tlv_data, tlv_size, PUBLIC_KEY_TAG, false);
-
-        if (public_key_tag_ptr) {
-          size_t length = ((size_t)public_key_tag_ptr[1] << 8 | public_key_tag_ptr[2]);
-          decode_public_key(self, public_key_tag_ptr + 3, length);
-
+        status = tlv_find_tag_and_decode(self, tlv_data, tlv_size, PUBLIC_KEY_TAG, false);
```

Comment:
There is a minor security/bug risk here when we combine both `status` and `public_key_found`. We should aim at using `status` for the SVI_TRY() scope below. Here we can put the call in the if explicitly.